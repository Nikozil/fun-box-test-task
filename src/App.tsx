import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import DotsListComponent from './components/DotsListComponent';
import Header from './components/Header';
import MapComponent from './components/MapComponent';
import NewDotForm from './components/NewDotForm';
import { reorder } from './lib/Functions';
import './styles/App.scss';

function App() {
  const [center, setCenter] = useState<Coordinates>([55.734108, 37.623915]);
  const [dotsList, setDotsList] = useState<Dot[]>([]);

  const handleSubmit = (name: DotName) => {
    const newDot = {
      id: uuidv4(),
      name: name,
      coordinates: center,
    };

    setDotsList((prevlist) => [...prevlist, newDot]);
  };

  const deleteDot = (id: DotId) => {
    setDotsList((prevlist) => prevlist.filter((dot: Dot) => dot.id !== id));
  };

  const permutationDots = (startIndex: number, endIndex: number) => {
    const newList = reorder(dotsList, startIndex, endIndex);
    setDotsList(newList);
  };

  return (
    <div className="app">
      <Header />
      <div className="interface">
        <NewDotForm handleSubmit={handleSubmit} />
        <DotsListComponent
          dotsList={dotsList}
          deleteDot={deleteDot}
          permutationDots={permutationDots}
        />
      </div>
      <div className="map">
        <MapComponent
          center={center}
          setCenter={setCenter}
          dotsList={dotsList}
          setDotsList={setDotsList}
        />
      </div>
    </div>
  );
}

export default App;

export interface Dot {
  id: DotId;
  name: DotName;
  coordinates: Coordinates;
}
export type DotId = string;
type DotName = string;
export type Coordinates = number[];
