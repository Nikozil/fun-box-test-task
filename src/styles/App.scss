@import url('https://fonts.googleapis.com/css2?family=Nunito&family=Roboto&display=swap');

.app {
  display: grid;
  grid-template-areas:
    'head head'
    'interface map';
  grid-template-rows: 60px 1fr;
  grid-template-columns: 300px 1fr;

  min-height: 100vh;
  min-width: 350px;

  font-family: 'Nunito', sans-serif;

  @media screen and (max-width: 767px) {
    grid-template-columns: 200px 1fr;
  }
}
.interface {
  grid-area: interface;
  background-color: #fff;
  padding: 10px;
}

.map {
  grid-area: map;
  background-color: #f7f7f7;
}
