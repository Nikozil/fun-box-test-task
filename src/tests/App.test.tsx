import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

describe('App tests', () => {
  test('renders  react correct', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
