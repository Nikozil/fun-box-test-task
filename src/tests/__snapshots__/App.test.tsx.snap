// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders  react correct 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <header
      class="header"
    >
      <div
        class="header__title"
      >
        Тестовое задание для FunBox 
      </div>
      <div
        class="header__author"
      >
        Выполнил Масляный Александр
      </div>
    </header>
    <div
      class="interface"
    >
      <form
        action="#"
        class="new-dot-form"
      >
        <label
          class="new-dot-form__label"
          for="newDot"
        >
          Введите метку:
        </label>
        <input
          autocomplete="off"
          class="new-dot-form__input"
          id="newDot"
          name="newDot"
          placeholder="Новая метка"
          type="text"
          value=""
        />
      </form>
      <ul
        class="dots-list__ul"
        data-rbd-droppable-context-id="0"
        data-rbd-droppable-id="droppable"
      />
    </div>
    <div
      class="map"
    />
  </div>
</DocumentFragment>
`;
